handling-forms/controlled-forms.md

```diff
@@ -0,0 +1,542 @@
+import { useState } from 'react';
+
+const NewStudentForm = () => {
+
+    const [formFields, setFormFields] = useState({
+        name: '',
+        email: ''
+    });
+```
+<!-- prettier-ignore-end -->
+
+Now she needs to make the input fields read from this state. Since `formFields` is an object, we'll use dot-notation to access the `name` and `email` values.
+
+<!-- prettier-ignore-start -->
+```js
+    return (
+        <form>
+            <div>
+                <label htmlFor="fullName">Name:</label>
+                <input
+                    name="fullName"
+                    value={formFields.name}/>
+            </div>
+            <div>
+                <label htmlFor="email">Email:</label>
+                <input
+                    name="email"
+                    value={formFields.email}/>
+            </div>
+            <input
+                type="submit"
+                value="Add Student"
+            />
+        </form>
+    );
+```
+<!-- prettier-ignore-end -->
+
+### Event Handling To Update `formFields`
+
+Sofia's next step is to ensure that changes in an input field affect the component state.
+
+She can create two event handlers, `onNameChange` and `onEmailChange`. These event handlers need to do two things:
+
+- Read the current value inside the input field
+- Update state to that current value
+
+<!-- prettier-ignore-start -->
+```js
+    const onNameChange = (event) => {
+        setFormFields({
+            ...formFields,
+            name: event.target.value
+        })
+    };
+
+    const onEmailChange = (event) => {
+        setFormFields({
+            ...formFields,
+            email: event.target.value
+        })
+    };
+```
+<!-- prettier-ignore-end -->
+
+In this example, we're using [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals) for a quick way to clone the original `formFields` object. Each event handler should add a specific key-value pair: `onNameChange` adds the key-value pair `name: event.target.value`, while `onEmailChange` adds `email`.
+
+Now, Sofia needs to make sure that the input fields use these event handlers when they're changed:
+
+<!-- prettier-ignore-start -->
+```js
+    return (
+        <form>
+            <div>
+                <label htmlFor="fullName">Name:</label>
+                <input
+                    name="fullName"
+                    value={formFields.name}
+                    onChange={onNameChange}/>
+            </div>
+            <div>
+                <label htmlFor="email">Email:</label>
+                <input name="email"
+                    value={formFields.email}
+                    onChange={onEmailChange}/>
+            </div>
+            <input
+                type="submit"
+                value="Add Student"
+            />
+        </form>
+    );
+```
+<!-- prettier-ignore-end -->
+
+We did it! 🎉🎉
+
+Sofia can verify that her form works as expected:
```

Comment:
Sofia did it! mention what about the behavior she can currently verify and set up for the next lesson
```suggestion
She did it! 🎉🎉

Sofia can verify that her form works as expected, allowing her to type in the name and email input fields. She can't _quite_ add the new student yet—she needs to handle submitting the form first—but she can confirm that her inputs are updating the component state in the React Dev Tools Components tab.
```